File: Week Assignment 6 using Python/1_queue_array.py
class Queue:
    def __init__(self):
        self.q = []

    def enqueue(self, x):
        self.q.append(x)

    def dequeue(self):
        if not self.q:
            return -1
        return self.q.pop(0)

    def front(self):
        return self.q[0] if self.q else -1

File: Week Assignment 6 using C/1_queue_array.c
#include <stdio.h>
#define MAX 1000
int queue[MAX];
int front = -1, rear = -1;

void enqueue(int x) {
    if (rear == MAX - 1) return;
    if (front == -1) front = 0;
    queue[++rear] = x;
}

void dequeue() {
    if (front == -1 || front > rear) return;
    front++;
}

int frontElement() {
    if (front == -1 || front > rear) return -1;
    return queue[front];
}
